tools/clang_format_check.py

```diff
@@ -106,7 +106,7 @@ def errors_from_replacements(file, replacements=[]):
 
     lines = [0]  # line index to character offset
     file_content = ""
-    for line in open(file, "r"):
+    for line in open(file, "r", encoding="utf-8"):
```

Comment:
did you get to the bottom of this? which file was responsible for the failure and why this flag is required?

Reply:
https://github.com/p4lang/PI/blob/master/src/p4info/p4info_struct.c#L58

I think it is the section sign here. I can also remove that and do not add the encoding.

Reply:
Adding the encoding is fine. I think ultimately it is because `open` means different things in Python 2 & 3. Speaking of which, if we still want to support both versions, should we use `io.open` here as suggested in https://docs.python.org/3/howto/pyporting.html ?

Reply:
So the way I understand this is that `io.open` has the same meaning in Python 2 that  `open` does in Python 3.
I just ran the Dockerfile with Python 2 instead of Python 3 and they all passed. Seems like open can also be used with that parameter in Python 2.  But no guarantees...